stingray/modeling/gpmodeling.py

```diff
@@ -37,10 +40,607 @@
+    f_max: float
+        The maximum frequency of the approximation grid.
+    n_approx_components: int
+        The number of components to use to approximate the power law
+        must be greater than 2, default 20
+    approximate_with: string
+        The type of kernel to use to approximate the power law power spectra
+        Default is "SHO"
+    """
+    f_c, a = _get_coefficients_approximation(
+        kernel_type,
+        kernel_params,
+        f_min,
+        f_max,
+        n_approx_components=n_approx_components,
+        approximate_with=approximate_with,
+    )
+    psd_SHO = SHO_power_spectrum(f, a[..., None], f_c[..., None]).sum(axis=0)
 
-def get_kernel(kernel_type, kernel_params):
+    psd_model = _psd_model(kernel_type, kernel_params)(f)
+    psd_model /= psd_model[..., 0, None]
+    return psd_model, psd_SHO
+
+
+def _approximate_powerlaw(
+    kernel_type, kernel_params, f_min, f_max, n_approx_components=20, approximate_with="SHO"
+):
+    """
+    Approximate the power law kernel with a sum of SHO kernels or a sum of DRW+SHO kernels.
+
+    Parameters
+    ----------
+    kernel_type: string
+        The type of kernel to be used for the Gaussian Process
+        Only designed for the following Power spectra ["PowL","DoubPowL]
+    kernel_params: dict
+        Dictionary containing the parameters for the kernel
+        Should contain the parameters for the selected kernel
+    f_min: float
+        The minimum frequency of the approximation grid.
+        Should be the lowest frequency of the power spectrum
+    f_max: float
+        The maximum frequency of the approximation grid.
+        Should be the highest frequency of the power spectrum
+    n_approx_components: int
+        The number of components to use to approximate the power law
+    approximate_with: string
+        The type of kernel to use to approximate the power law power spectra
+        Default is "SHO"
+    """
+    spectral_points, spectral_coefficients = _get_coefficients_approximation(
+        kernel_type,
+        kernel_params,
+        f_min,
+        f_max,
+        n_approx_components=n_approx_components,
+        approximate_with=approximate_with,
+    )
+
+    if approximate_with == "SHO":
+        amplitudes = (
+            spectral_coefficients
+            * spectral_points
+            * kernel_params["variance"]
+            / jnp.sum(spectral_coefficients * spectral_points)
+        )
+
+        kernel = amplitudes[0] * kernels.quasisep.SHO(
+            quality=1 / jnp.sqrt(2), omega=2 * jnp.pi * spectral_points[0]
+        )
+        for j in range(1, n_approx_components):
+            kernel += amplitudes[j] * kernels.quasisep.SHO(
+                quality=1 / jnp.sqrt(2), omega=2 * jnp.pi * spectral_points[j]
+            )
+        return kernel
+    else:
+        raise NotImplementedError(f"Approximation {approximate_with} not implemented")
+
+
+def _psd_model(kernel_type, kernel_params):
```

Comment:
Curious why this has a "_" in front of it? Do we not expect users to want to get the power spectrum model, or is there another function that fulfils this task?